pkg/api/validate/openshiftcluster_validatedynamic.go

```diff
@@ -287,3 +294,20 @@ func (dv *openShiftClusterDynamicValidator) validateVnet(ctx context.Context, vn
 
 	return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidLinkedVNet, "", "The provided vnet '%s' is invalid: custom DNS servers are not supported.", vnetID)
 }
+
+func (dv *openShiftClusterDynamicValidator) validateProviders(ctx context.Context, providerClient resources.ProvidersClient) error {
+	providers := []string{
+		"Microsoft.Storage",
+	}
+	for _, provider := range providers {
+		result, err := providerClient.Get(ctx, provider, "")
+		if err != nil {
+			return err
+		}
+		if *result.RegistrationState != "Registered" {
```

Comment:
I wasn't sure what this result would be other than what was returned via the python cli.  I'll create a smaller client version to verify.

Reply:
```
go run resclient/cmd/main.go
// 	fmt.Printf("%v\n", to.String(result.RegistrationState))
Registered
```

Reply:
lgtm